libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/ConwayFeatures.hs

```diff
@@ -0,0 +1,379 @@
+
+  assertEqual "govState after proposal" govState0 expectedGovernance0
+
+  let
+    voteTx = txFromTestCaseData pf (vote pf govActionId)
+    gas = govActionStateWithVote (newConstitutionProposal pf) (stakePoolKeyHash pf) VoteYes
+    expectedGovState1 = ConwayGovState $ Map.fromList [(govActionId, gas)]
+    expectedGovernance1 = ConwayGovernance expectedGovState1 (initialGovernance ^. cgRatifyL)
+    eitherLedgerState1 = runLEDGER (LEDGER pf) ledgerState0 (pp pf) (trustMeP pf True voteTx)
+    ledgerState1@(LedgerState (UTxOState _ _ _ govState1 _) _) = fromRight (error "error running LEDGER when voting") eitherLedgerState1
+
+  assertEqual "govState after vote" govState1 expectedGovernance1
+
+  let
+    epochState0 = (def :: EpochState era) & esPpL .~ pp pf & esLStateL .~ ledgerState1
+    poolDistr =
+      PoolDistr
+        ( Map.fromList
+            [
+              ( stakePoolKeyHash pf
+              , IndividualPoolStake
+                  (spoThreshold * 2)
+                  (vrfKeyHash @(EraCrypto era))
+              )
+            ]
+        )
+    eitherEpochState1 = runEPOCH (EPOCH pf) epochState0 (EpochNo 2) poolDistr
+    epochState1 = fromRight (error "oops in runEpoch") eitherEpochState1
+    ledgerState2 = epochState1 ^. esLStateL
+    constitution = (ensConstitution . rsEnactState) (epochState1 ^. esLStateL . lsUTxOStateL . utxosGovernanceL . cgRatifyL)
+  assertEqual "constitution after enactment" constitution (proposedConstitution @era)
+
+  let
+    secondProposalTx = txFromTestCaseData pf (secondProposal pf govActionId)
+    secondGovActionId = GovernanceActionId (txid (secondProposalTx ^. bodyTxL)) (GovernanceActionIx 0)
+    expectedGovState2 = ConwayGovState $ Map.fromList [(secondGovActionId, govActionState (anotherConstitutionProposal pf))]
+    expectedGovernance2 = ConwayGovernance expectedGovState2 (ledgerState2 ^. lsUTxOStateL . utxosGovernanceL . cgRatifyL)
+    eitherLedgerState3 = runLEDGER (LEDGER pf) ledgerState2 (pp pf) (trustMeP pf True secondProposalTx)
+    ledgerState3@(LedgerState (UTxOState _ _ _ govState2 _) _) = fromRight (error "error running LEDGER when proposing") eitherLedgerState3
+
+  assertEqual "govState after second proposal" govState2 expectedGovernance2
+
+  let
+    epochState2 = epochState1 & esLStateL .~ ledgerState3
+    eitherEpochState2 = runEPOCH (EPOCH pf) epochState2 (EpochNo 2) poolDistr
+    epochState3 = fromRight (error "oops in runEpoch") eitherEpochState2
+    constitution1 = (ensConstitution . rsEnactState) (epochState3 ^. esLStateL . lsUTxOStateL . utxosGovernanceL . cgRatifyL)
+  assertEqual "constitution after enactment after no votes" constitution1 (proposedConstitution @era)
+
+  let (gId, gas') = head . foldr (:) [] $ rsFuture (epochState3 ^. esLStateL . lsUTxOStateL . utxosGovernanceL . cgRatifyL)
+  assertEqual "un-enacted govAction is recorded in rsFuture" (gId, gas') (secondGovActionId, govActionState (anotherConstitutionProposal pf))
```

Comment:
@Soupstraw  This is the check that I added after we talked, but it seems quite superficial - do you have a better idea how to test deposit returns?